src/stimfit/gui/graph.cpp

```diff
@@ -2152,6 +2163,50 @@ void wxStfGraph::OnDown() {
     Refresh();
 }
 
+void wxStfGraph::ChanScroll(int direction) {
+    /* on Control + cursor press, adjust the active channel
+       up or down. 
+    */
+    
+    int reference_ch = Doc()->GetSecChIndex();  
+    
+    int new_channel = Doc()->GetCurChIndex() + direction;
+    
+    
+    //------------------------------------------
+    // Rollover conditions
+    //------------------------------------------
+    if (new_channel >= Doc()->size() - 1) {
+    	// Rollover to start if channel out of range
+	new_channel = 0;
+    }
+    if (new_channel < 0) {
+	// Rollover to end if channel out of range
+	new_channel = Doc()->size() - 1;
+    } 
+    
+   if (new_channel == reference_ch) {
+	// Skip the reference channel
+	new_channel += direction;
```

Comment:
Is there like a debugging tool that I should be using for this? I can't seem to get back to a state with no errors and a refactored function.